internal/middleware/authentication.go

```diff
@@ -67,60 +70,38 @@ func (m AuthenticationMiddleware) TokenAuthentication(c *gin.Context) {
 		c.Abort()
 		return
 	} else {
-		c.Set("user", u)
+		c.Set("user", user)
 		c.Next()
 	}
 }
 
 func parseRequest(request *http.Request, key *rsa.PublicKey) (*model.User, error) {
 	token, err := jwt.ParseRequest(
 		request,
-		jwt.WithValidate(true),
-		jwt.WithVerify(jwa.RS256, key),
+		jwt.WithKey(jwa.RS256, key),
+		jwt.WithHeaderKey("Authorization"),
+		jwt.WithCookieKey("accessToken"),
+		jwt.WithCookieKey("refreshToken"),
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	userData, ok := token.Get("user")
-	if !ok {
-		return nil, errors.New("user not found in claims")
-	}
-
-	return extractUser(userData)
+	return extractUser(token)
 }
 
-func extractUser(userData any) (*model.User, error) {
-	userMap, ok := userData.(map[string]any)
+func extractUser(token jwt.Token) (*model.User, error) {
+	userData, ok := token.Get("user")
 	if !ok {
-		return nil, errors.New("failed to parse user data")
+		return nil, errors.New("user not found in claims")
 	}
```

Comment:
Should we keep the error text prefix be "failed to ..." here for consistency?

Reply:
No :slightly_smiling_face: